core/handlers/validation/builtin/v12/validation_logic_test.go

```diff
@@ -2272,10 +2272,14 @@ var mockMSPIDGetter = func(cid string) []string {
 }
 
 func TestMain(m *testing.M) {
+	code := -1
+	defer func() {
+		os.Exit(code)
+	}()
```

Comment:
i would probably wrap this into `testMain(m *testing.M) int { defer ... }` which returns exit code, and calls `os.Exit(testMain(m))` from `TestMain`. But it's effectively the same